static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -301,57 +314,213 @@ class SpanTreeModel {
+                      ? this.showSpanGroup
+                      : showSpanGroup
+                    : false,
+                  addTraceBounds,
+                  removeTraceBounds,
+                  isAutogroupSiblingFeatureEnabled,
+                })
+              );
+
+              acc.previousSiblingEndTimestamp = spanModel.span.timestamp;
+            });
+
+            return acc;
+          }
+
+          // NOTE: I am making the assumption here that grouped sibling spans will not have children.
+          // By making this assumption, I can immediately wrap the grouped spans here without having
+          // to recursively traverse them.
+
+          // This may not be the case, and needs to be looked into later
+
+          const wrappedSiblings: EnhancedSpan[] = group.map((spanModel, index) => {
+            const enhancedSibling: EnhancedSpan = {
+              type: 'span',
+              span: spanModel.span,
+              numOfSpanChildren: 0,
+              treeDepth: treeDepth + 1,
+              isLastSibling:
+                index === group.length - 1 &&
+                groupIndex === groupedDescendants.length - 1,
+              isFirstSiblingOfGroup: index === 0,
+              continuingTreeDepths: descendantContinuingTreeDepths,
+              fetchEmbeddedChildrenState: spanModel.fetchEmbeddedChildrenState,
+              showEmbeddedChildren: spanModel.showEmbeddedChildren,
+              toggleEmbeddedChildren: spanModel.toggleEmbeddedChildren({
+                addTraceBounds,
+                removeTraceBounds,
+              }),
+              toggleSpanGroup: undefined,
+              toggleSiblingSpanGroup:
+                index === 0 ? this.toggleSiblingSpanGroup : undefined,
+            };
+
+            return enhancedSibling;
+          });
+
+          if (this.isSpanFilteredOut(props, group[0])) {
+            group.forEach(spanModel =>
+              acc.descendants.push({
+                type: 'filtered_out',
+                span: spanModel.span,
+              })
+            );
+            return acc;
+          }
+
+          const groupedSiblingsSpan: EnhancedProcessedSpanType = {
+            type: 'span_group_sibling',
+            span: this.span,
+            treeDepth: treeDepth + 1,
+            continuingTreeDepths,
+            spanGrouping: wrappedSiblings,
+            toggleSpanGroup: undefined,
+            showSpanGroup,
+            isLastSibling: groupIndex === groupedDescendants.length - 1,
+            toggleSiblingSpanGroup: this.toggleSiblingSpanGroup,
+          };
+
+          // Check if the group is currently expanded or not
+          const key = `${group[0].span.op}.${group[0].span.description}`;
+          if (this.ungroupedSiblings.has(key)) {
+            acc.descendants.push(...wrappedSiblings);
+            return acc;
+          }
+
+          acc.descendants.push(groupedSiblingsSpan);
+
+          return acc;
```

Comment:
@0Calories Some preliminary feedback. `acc.previousSiblingEndTimestamp` needs to be set here to the end timestamp of the last span in either `wrappedSiblings` or `groupedSiblingsSpan.spanGrouping`.